test/jtreg_test_thread_factory/src/share/classes/Virtual.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.util.concurrent.ThreadFactory;
+
+public class Virtual implements ThreadFactory {
+
+    static {
+        // This property is used by ProcessTools and some tests
+        try {
+            System.setProperty("main.wrapper", "Virtual");
+        } catch (Throwable t) {
+            // might be thrown by security manager
+        }
+    }
+
+    public Virtual() {
+    }
+
+
+    @Override
+    public Thread newThread(Runnable task) {
+        try {
+            return Thread.ofVirtual().factory().newThread(task);
```

Comment:
As far as I can see, none of these method calls throw a checked exception. Should we perhaps remove this try/catch block and let any runtime exception that gets thrown be propagated as-is instead of wrapping it in a `RuntimException`?

Reply:
It's also a bit strange that it creates a new ThreadFactory each time, I had expected to be a wrapper around one ThreadFactory. The effect is the same, just a bit strange.

(I suspect the try-catch dates from when it called Thread.ofVirtual with reflection).